include/boost/astronomy/coordinate/base_ecliptic_frame.hpp

```diff
@@ -2,185 +2,333 @@
+                typename OtherRepresentation::quantity1,
+                typename OtherRepresentation::quantity2,
+                typename OtherRepresentation::quantity3
+            >(representation_data);
+        this->data = rep_temp;
+
+        auto dif_temp = make_spherical_coslat_differential
+            <
+                typename Differential::type,
+                typename Differential::quantity1,
+                typename Differential::quantity2,
+                typename Differential::quantity3,
+                typename OtherDifferential::type,
+                typename OtherDifferential::quantity1,
+                typename OtherDifferential::quantity2,
+                typename OtherDifferential::quantity3
+            >(diff);
+        this->motion = dif_temp;
+    }
+
+    //!returns latitude component of the coordinate
+    typename Representation::quantity1 get_lat() const
+    {
+        return this->data.get_lat();
+    }
+
+    //!returns longitude component of the coordinate
+    typename Representation::quantity2 get_lon() const
+    {
+        return this->data.get_lon();
+    }
+
+    //!returns distance component of the coordinate
+    typename Representation::quantity3 get_distance() const
+    {
+        return this->data.get_dist();
+    }
+
+    //!returns the (lat, lon, dist) in the form of tuple
+    std::tuple<typename Representation::quantity1, typename Representation::quantity2,
+        typename Representation::quantity3> get_lat_lon_dist() const
+    {
+        return this->data.get_lat_lon_dist();
+    }
+
+    //!returns proper motion in latitude
+    typename Differential::quantity1 get_pm_lat() const
+    {
+        return this->motion.get_dlat();
+    }
+
+    //!returns proper motion in longitude including cos(lat)
+    typename Differential::quantity2 get_pm_lon_coslat() const
+    {
+        return this->motion.get_dlon_coslat();
+    }
+
+    //!returns radial_velocity
+    typename Differential::quantity3 get_radial_velocity() const
+    {
+        return this->motion.get_ddist();
+    }
+
+    //!returns the proper motion in form of tuple
+    std::tuple<typename Differential::quantity1, typename Differential::quantity2,
+        typename Differential::quantity3> get_pm_lat_lon_radial() const
+    {
+        return this->motion.get_dlat_dlon_coslat_ddist();
+    }
 
+    //!sets value of latitude component of the coordinate
+    void set_lat(typename Representation::quantity1 const& lat)
+    {
+        this->data.set_lat(lat);
+    }
+
+    //!sets value of longitude component of the coordinate
+    void set_lon(typename Representation::quantity2 const& lon)
+    {
+        this->data.set_lon(lon);
+    }
+
+    //!sets value of distance component of the coordinate
+    void set_distance(typename Representation::quantity3 const& distance)
+    {
+        this->data.set_dist(distance);
+    }
+
+    //!sets values of all component of the coordinate
+    void set_alt_az_dist
```

Comment:
the function name is wrong `alt` and `az` coordinates are not used in the ecliptic frame.